keps/sig-storage/20191125-bucket-provisioning.md

```diff
@@ -222,25 +228,29 @@ Metadata:
   - cosi.io/finalizer [3]
 spec:
   bucketClassName: [4]
-  releasePolicy: {"Delete", "Retain"} [5]
-  bucketConfig: map[string]string [6]
-  bucketRef: [7]
+  supportedProtocols: [5]
+  releasePolicy: {"Delete", "Retain"} [6]
+  bucketConfig: map[string]string [7]
+  bucketRef: [8]
     name:
     namespace:
-  secretName: [8]
+  secretRef: [9]
+  accessMode: {"rw", "ro"} [10]
 status:
-  bucketMetaData: <map[string]string> [9]
-  phase: {"Bound", "Released", "Failed", "Errored"} [10]
+  bucketAttributes: <map[string]string> [11]
+  phase: {"Bound", "Released", "Failed", "Errored"} [12]
   conditions:
 ```
 1. `name`: Generated in the pattern of `“bucket-”<BUCKET-NAMESPACE>"-"<BUCKET-NAME>`
 1. `labels`: central controller adds the label to its managed resources for easy CLI GET ops.  Value is the driver name returned by GetDriverInfo() rpc*.
 1. `finalizers`: COSI controller adds the finalizer to defer Bucket deletion until backend deletion ops succeed.
 1. `bucketClassName`: Name of the associated BucketClass
+1. `supportedProtocols`:  String array of protocols (e.g. s3, gcs, swift, etc.) supported by the associated object store.
 1. `releasePolicy`: the release policy defined in the associated BucketClass. (see [BucketClass](#BucketClass) for more information)
-1. `bucketConfig`: a string:string map of driver defined key-value pairs
+1. `bucketAttributes`: a string:string map of driver defined key-value pairs
 1. `bucketRef`: the name & namespace of the bound Bucket.
-1. `secretName`: the name of the sidecar-generated secret. It's namespace is assumed `cosi-system`.
+1. `secretRef`: the name of the sidecar-generated secret. It's namespace is assumed `cosi-system`.
```

Comment:
why are we calling this a `ref` when it is only a name?

Reply:
The secretRef reflects that it's an actual `type secretRef struct` object.  It's important because the BucketContent is cluster-scoped.  Depending on how the community falls on the driver namespaces, this is important as we want to be able to target secrets accurately.

Reply:
If the driver's secret's ns is not fixed then I agree. But if the ns is always known then it seems like name alone is sufficient. I haven't checked to see what is done in other CRs so the examples I'm thinking of our cluster scoped, like SC.

Reply:
Right, it's dependent on where the community settles on this.